####################################################################################################
This patch file is from Qualcomm Technologies, Inc. and is provided under the following license:

Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
SPDX-License-Identifier: BSD-3-Clause-Clear
####################################################################################################
From cc81f372ba69c73eb2b8903f047ddc7a380c8148 Mon Sep 17 00:00:00 2001
From: jiafwu <jiafwu@qti.qualcomm.com>
Date: Tue, 10 Jun 2025 10:54:21 +0800
Subject: [PATCH 2/2] Fix the compilation error of missing absl::StrCat
 function

Upstream-Status: Inappropriate [Qualcomm specific]

Signed-off-by: jiafwu <jiafwu@qti.qualcomm.com>
---
 tensorflow/lite/delegates/gpu/cl/cl_operation.h                  | 1 +
 .../lite/delegates/gpu/common/task/qcom_thin_filter_desc.cc      | 1 +
 .../delegates/gpu/common/tasks/special/thin_pointwise_fuser.cc   | 1 +
 3 files changed, 3 insertions(+)

diff --git a/tensorflow/lite/delegates/gpu/cl/cl_operation.h b/tensorflow/lite/delegates/gpu/cl/cl_operation.h
index c9192830a02..772269c003f 100644
--- a/tensorflow/lite/delegates/gpu/cl/cl_operation.h
+++ b/tensorflow/lite/delegates/gpu/cl/cl_operation.h
@@ -22,6 +22,7 @@ limitations under the License.
 #include <utility>
 #include <vector>
 
+#include "absl/strings/str_cat.h"
 #include "tensorflow/lite/delegates/gpu/cl/cl_arguments.h"
 #include "tensorflow/lite/delegates/gpu/cl/cl_command_queue.h"
 #include "tensorflow/lite/delegates/gpu/cl/cl_context.h"
diff --git a/tensorflow/lite/delegates/gpu/common/task/qcom_thin_filter_desc.cc b/tensorflow/lite/delegates/gpu/common/task/qcom_thin_filter_desc.cc
index 03909221f99..ffcf1fb2cdf 100644
--- a/tensorflow/lite/delegates/gpu/common/task/qcom_thin_filter_desc.cc
+++ b/tensorflow/lite/delegates/gpu/common/task/qcom_thin_filter_desc.cc
@@ -18,6 +18,7 @@ limitations under the License.
 #include <string>
 #include <vector>
 
+#include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 
 namespace tflite {
diff --git a/tensorflow/lite/delegates/gpu/common/tasks/special/thin_pointwise_fuser.cc b/tensorflow/lite/delegates/gpu/common/tasks/special/thin_pointwise_fuser.cc
index 03fd1292827..923aaf53efa 100644
--- a/tensorflow/lite/delegates/gpu/common/tasks/special/thin_pointwise_fuser.cc
+++ b/tensorflow/lite/delegates/gpu/common/tasks/special/thin_pointwise_fuser.cc
@@ -24,6 +24,7 @@ limitations under the License.
 #include <utility>
 #include <vector>
 
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_replace.h"
 #include "tensorflow/lite/delegates/gpu/common/flops_util.h"
 #include "tensorflow/lite/delegates/gpu/common/model.h"
-- 
2.25.1

